### PR TITLE
veikk-linux-driver-gui: init at 2.0

### PR DESCRIPTION
**Pull request by nicbk**
Add configuration utility to configure VEIKK brand digitizers

###### Motivation for this change
Add configuration utility to configure VEIKK brand digitizers

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by nicbk** [comment]
Original driver has been submitted here -> #125856

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -31806,4 +31806,9 @@ in
   zthrottle = callPackage ../tools/misc/zthrottle { };
 
   zktree = callPackage ../applications/misc/zktree {};
+
+  veikk-linux-driver-gui = callPackage ../os-specific/linux/veikk/gui {
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:31810]: Please sort this into a fitting category and sort there alphabetically.

File: pkgs/os-specific/linux/veikk/gui/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  version = "v2.0";
+
+  src = fetchFromGitHub {
+    owner = "jlam55555";
+    repo = name;
+    rev = version;
+    sha256 = "02g1q79kwjlzg95w38a1d7nxvcry8xcsvhax2js4c7xqvzhkki5j";
+  };
+
+  nativeBuildInputs = [ qmake gnumake ];
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/veikk/gui/default.nix:14]:
```suggestion
  nativeBuildInputs = [ qmake ];
```
already part of stdenv.

File: pkgs/os-specific/linux/veikk/gui/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+  buildPhase = ''
+    qmake
+    make all clean
+  '';
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/veikk/gui/default.nix:19]:
```suggestion
  postBuild = ''
    make all clean
  '';
```

File: pkgs/os-specific/linux/veikk/gui/default.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
+{ lib, mkDerivation, fetchFromGitHub, gnumake, qmake }:
+
+mkDerivation rec {
+  name = "veikk-linux-driver-gui";
+  version = "v2.0";
+
+  src = fetchFromGitHub {
+    owner = "jlam55555";
+    repo = name;
+    rev = version;
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/veikk/gui/default.nix:10]:
```suggestion
    rev = "v${version}";
```

File: pkgs/os-specific/linux/veikk/gui/default.nix [hunk 5]
```diff
@@ -0,0 +1,33 @@
+{ lib, mkDerivation, fetchFromGitHub, gnumake, qmake }:
+
+mkDerivation rec {
+  name = "veikk-linux-driver-gui";
+  version = "v2.0";
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/veikk/gui/default.nix:5]:
```suggestion
  version = "2.0";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (nicbk) [reply pkgs/top-level/all-packages.nix]: Moved into pkgs/tools/misc/

File: pkgs/top-level/all-packages.nix [hunk 6]
```diff
@@ -870,6 +870,11 @@ in
 
   tnat64 = callPackage ../tools/networking/tnat64 { };
 
+  veikk-linux-driver-gui = callPackage ../tools/misc/veikk-linux-driver-gui {
+    inherit (qt5) qmake;
+    inherit (qt5) mkDerivation;
+  };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:876]:
```suggestion
  veikk-linux-driver-gui = libsForQt5.callPackage ../tools/misc/veikk-linux-driver-gui { };
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please squash all package related commits together except the maintainer one.

**Pull request merged**